tests/test_pipeline.py

```diff
@@ -46,6 +50,38 @@ def _test_image_to_text(self, model, expected_result):
                 output = generator(image, generate_kwargs=generate_kwargs)
             self.assertTrue(output[0]["generated_text"].startswith(expected_result))
 
+    def _test_text_to_speech(self, model, expected_sample_rate):
+        adapt_transformers_to_gaudi()
+        MODEL_DTYPE_LIST = [torch.bfloat16, torch.float32]
+        text = "hello, the dog is cooler"
+        for model_dtype in MODEL_DTYPE_LIST:
+            generator = pipeline(
+                "text-to-speech",
+                model=model,
+                torch_dtype=model_dtype,
+                device="hpu",
+            )
+            forward_params = None
+            if generator.model.config.model_type == "speecht5":
+                embeddings_dataset = load_dataset("Matthijs/cmu-arctic-xvectors", split="validation")
+                speaker_embedding = torch.tensor(embeddings_dataset[7306]["xvector"]).unsqueeze(0).to("hpu")
+                forward_params = {"speaker_embeddings": speaker_embedding}
+            if generator.model.config.model_type == "seamless_m4t":
+                forward_params = {"tgt_lang": "eng"}
+
+            generate_kwargs = None
+            if generator.model.can_generate():
+                generate_kwargs = {"lazy_mode": True, "ignore_eos": False, "hpu_graphs": True}
+
+            generator.model = wrap_in_hpu_graph(generator.model)
+            with torch.autocast(
+                "hpu", torch.bfloat16, enabled=(model_dtype == torch.bfloat16)
+            ), torch.no_grad(), torch.inference_mode():
```

Comment:
`torch.no_grad()` and `torch.inference_mode()` are a bit redundant no?
I think we can just keep `torch.inference_mode()`.